R/LoadCanada.R

```diff
@@ -17,97 +17,101 @@
-
-  # Data aggregated by the COVID-19 Canada Open Data Working Group https://github.com/ccodwg/Covid19Canada
-  # Berry I, Soucy J-PR, Tuite A, Fisman D. Open access epidemiologic data and an interactive dashboard to monitor the COVID-19 outbreak in Canada. CMAJ. 2020 Apr 14;192(15):E420. doi:  https://doi.org/10.1503/cmaj.75262.
-
-
-  CANADADATA <- vroom::vroom("https://raw.githubusercontent.com/ccodwg/Covid19Canada/master/timeseries_hr/cases_timeseries_hr.csv")
-
-  # pop_canada<- read.csv("https://raw.githubusercontent.com/ishaberry/Covid19Canada/master/other/hr_map.csv",encoding="UTF-8")#contains pop and health region ID codes
-
-  # shrink this data file
-  DATES <- as.Date(CANADADATA$date_report, format = "%d-%m-%Y")
-  DatIND <- which(DATES < "2021-10-01")
-  CANDATASMALLER <- CANADADATA[-DatIND, ]
-
-  # Assemble the data required to calculate risk scores and match health region ID's
-  CANADARISK <- c()
-  count <- 1
-  CPro <- unique(pop_canada$province)
-  for (aa in 1:length(CPro)) {
-    # cat("aa:",aa,"\n")
-    subsetCANL <- pop_canada[pop_canada$province == CPro[aa], ]
-    subsetCAND <- CANDATASMALLER[CANDATASMALLER$province == CPro[aa], ]
-    CHR <- unique(subsetCANL$health_region)
-    for (bb in 1:length(CHR)) {
-      # cat("bb:",bb,"\n")
-      subset2CAND <- subsetCAND[subsetCAND$health_region == CHR[bb], ]
-      subset2CANL <- subsetCANL[subsetCANL$health_region == CHR[bb], ]
-      CANDATES <- as.Date(subset2CAND$date_report, format = "%d-%m-%Y")
-
-      MD <- max(CANDATES)
-      MC <- subset2CAND$cumulative_cases[which(CANDATES == MD)]
-      LD <- MD - 14
-      LC <- subset2CAND$cumulative_cases[which(CANDATES == LD)]
-
-      CANADARISK$province[count] <- CPro[aa]
-      CANADARISK$health_region[count] <- CHR[bb]
-      CANADARISK$HR_UID[count] <- as.character(subset2CANL$HR_UID)
-      CANADARISK$pop[count] <- subset2CANL$pop
-      CANADARISK$province_full[count] <- subset2CANL$province_full
-      CANADARISK$DateReport[count] <- as.character(MD)
-      CANADARISK$CaseDiff[count] <- (MC - LC) / 14 * 10
-      CANADARISK$pInf[count] <- CANADARISK$CaseDiff[count] / CANADARISK$pop[count]
-      count <- count + 1
+    #Load geometry and population data
+    pop_canada <-NULL
+    utils::data("geomCanada", "pop_canada", envir = environment())
+    geomCanada <- sf::st_as_sf(geomCanada)
+    
+    # Data aggregated by the COVID-19 Canada Open Data Working Group https://github.com/ccodwg/Covid19Canada
+    # Berry I, Soucy J-PR, Tuite A, Fisman D. Open access epidemiologic data and an interactive dashboard to monitor the COVID-19 outbreak in Canada. CMAJ. 2020 Apr 14;192(15):E420. doi:  https://doi.org/10.1503/cmaj.75262.
+    
+    
+    #CANADADATA <- vroom::vroom("https://raw.githubusercontent.com/ccodwg/Covid19Canada/master/timeseries_hr/cases_timeseries_hr.csv")
+    CANADADATA <- vroom::vroom("https://raw.githubusercontent.com/ccodwg/CovidTimelineCanada/main/data/hr/cases_hr.csv")   
```

Comment:
Thanks @kaitejohnson!

Reply:
Whoops yeah looks like it's an issue with the matching of non-reported, let me try to fix!

Reply:
Ok, try now! The github pop_canada which was commented out, seems to have different health region IDs than the data canada is prociving, and I think its the github that is wrong (since the others are all 4 digits). For some reason, my load of the pop_canada object from the package isn't working so I can't really troubleshoot, but I think this should work.

Confirming that you ignore cases that are assigned not reported right? Since you do an inner join anyway?

Reply:
Ok let me know, I think the issue was just around the 'pop_canada` object for `SK` province, at least on my end